runtime/lua/vim/fs.lua

```diff
@@ -629,6 +629,10 @@ function M.normalize(path, opts)
       return prefix .. path
     end
 
+    if prefix:match('%w:') then
+      prefix = string.upper(prefix)
+    end
+
     -- Remove extraneous slashes from the prefix
     prefix = prefix:gsub('/+', '/')
```

Comment:
```suggestion
    -- Ensure capital drive and remove extraneous slashes from the prefix
    prefix = prefix:gsub('^%a:', string.upper):gsub('/+', '/')
```

A neat little trick is that `gsub` allows function as replacement, which is appropriate here.